src/main/java/net/imglib2/realtransform/ExplicitInvertibleRealTransform.java

```diff
@@ -0,0 +1,93 @@
+package net.imglib2.realtransform;
+
+import net.imglib2.RealLocalizable;
+import net.imglib2.RealPositionable;
+
+/**
+ * An {@link InvertibleRealTransform} for which the forward and inverse
+ * transforms are explicitly given as {@link RealTransform}s. Consistency is not
+ * internally enforced.
+ * 
+ * @author John Bogovic
+ * @author Stephan Saalfeld
+ *
+ */
+public class ExplicitInvertibleRealTransform implements InvertibleRealTransform
+{
+	private final RealTransform forwardTransform;
+
+	private final RealTransform inverseTransform;
+
+	private final ExplicitInvertibleRealTransform inverse;
+
+	/**
+	 * Creates a new ExplicitInvertibleRealTransform.
+	 *  
+	 * @param forwardTransform the transform defining the forward direction
+	 * @param inverseTransform the trasnform defining the inverse direction
+	 */
+	public ExplicitInvertibleRealTransform( final RealTransform forwardTransform, final RealTransform inverseTransform )
+	{
+		this.forwardTransform = forwardTransform;
+		this.inverseTransform = inverseTransform;
+		this.inverse = new ExplicitInvertibleRealTransform( inverseTransform, forwardTransform, this );
+		// TODO explicitly check that source and target dimensions for forward
+		// and inverse transforms are compatible?
+	}
+
+	private ExplicitInvertibleRealTransform( final RealTransform forwardTransform, final RealTransform inverseTransform, final ExplicitInvertibleRealTransform inverse )
```

Comment:
How about avoiding possible inconsistency
```java
private ExplicitInvertibleRealTransform( final ExplicitInvertibleRealTransform inverse )
	{
		this.forwardTransform = inverse.inverseTransform;
		this.inverseTransform = inverse.forwardTransform;
		this.inverse = inverse;
	}
```

Reply:
@bogovicj Can you please do it?  GitHub makes it incredibly hard to contribute to pull requests from private repositories.

Reply:
@bogovicj forget about it, I made it.  Still hate that they do not make the source clickable...

Reply:
oops, thanks for the reminder, on it

Reply:
I see now, thanks.